File: swiggy-backend/src/main/java/com/swiggy/swiggy_backend/service/FavoriteService.java
package com.swiggy.swiggy_backend.service;

import java.util.List;

import com.swiggy.swiggy_backend.dto.FavoriteResponse;

public interface FavoriteService {

    void addFavorite(Long restaurantId);

    void removeFavorite(Long restaurantId);

    List<FavoriteResponse> getFavorites();
}

File: swiggy-frontend/src/services/favoriteService.js

import api from "./api";

export const addFavorite = (restaurantId) => {
    return api.post(`/favorites/${restaurantId}`);
};

export const getFavorites = () => {
    return api.get("/favorites");
};

export const removeFavorite = (restaurantId) => {
    return api.delete(`/favorites/${restaurantId}`);
};